pkgs/tools/misc/qmk/default.nix

```diff
@@ -27,13 +39,23 @@ buildPythonApplication rec {
     appdirs
     argcomplete
     colorama
-    dotty-dict
+    #dotty-dict # See https://pypi.org/project/qmk-dotty-dict/
+    qmk-dotty-dict
     hid
     hjson
     jsonschema
     milc
     pygments
+    pyrsistent
     pyusb
+  ] ++ [ # Binaries need to be in the path so this is in propagatedBuildInputs
+    avrdude
+    dfu-programmer
+    dfu-util
+    gcc-arm-embedded
+    pkgsCross.avr.buildPackages.binutils
```

Comment:
```suggestion
    pkgsCross.avr.buildPackages.binutils
    pkgsCross.avr.buildPackages.binutils.bintools
```
See my comment: https://github.com/NixOS/nixpkgs/pull/130847#issuecomment-899090604 — AVR firmwares don't compile without this.